### PR TITLE
Simulator/update level ratio price

### PR DESCRIPTION
**Pull request by Blinningjr**
Worked on:
* Added controller for the market in web-server.
* Added api for market in the simulator
* Added so the production level can be set in the simulator.
* Added so the market price can be set in the simulator.
* Added so the marketRatio can ve set in the simulator.

### CODE REVIEW

File: simulator/src/api/power-plantapi.ts [hunk 1]
```diff
@@ -70,20 +70,52 @@ router.delete('/my', ensureAuthenticated('manager'), async (req: express.Request
 
 
 /**
- * Get the power plants markets suggested price.
+ * Update power plant production level.
  */
-router.get('/market/suggested-price', ensureAuthenticated('manager'), 
-    (req: express.Request, res: express.Response) => {
+router.post('/update/production/level', ensureAuthenticated('manager'),
```
Comment (Aleman778) [inline simulator/src/api/power-plantapi.ts]: I recommend removing verbs e.g. update from url and instead use put request.

File: simulator/src/api/power-plantapi.ts [hunk 2]
```diff
@@ -70,20 +70,52 @@ router.delete('/my', ensureAuthenticated('manager'), async (req: express.Request
...
+        console.log("[Power PlantAPI] Failed to find power plant with id " + req.userId + ".");
     }
     return res.status(400).send("Whoops! We failed to find your power plant, please try again later.");
 });
 
+
+/**
+ * Update power plant market ratio.
+ */
+router.post('/update/market-ratio', ensureAuthenticated('manager'),
```
Comment (Aleman778) [inline simulator/src/api/power-plantapi.ts]: Same here just PUT /api/power-plant/market-ratio

File: simulator/src/app.ts [hunk 3]
```diff
@@ -52,6 +53,9 @@ app.use('/api/house', houseapi);
 // Mount REST api route /api/power-plant to power-plantapi
 app.use('/api/power-plant', powerPlantApi);
 
+// Mount REST api route /api/market to marketapi
+app.use('/api/market', marketApi);
+
```
Comment (Aleman778) [inline simulator/src/app.ts]: This is incompatible with my branch, either wait until my branch is merged to master and update then.

Comment (Blinningjr) [reply simulator/src/app.ts]: Why should this branch wait? I can't think of a good reason for waiting for the other pull request.

File: simulator/src/api/marketapi.ts [hunk 4]
```diff
@@ -13,7 +13,7 @@ var router = express.Router();
 /**
  * Get current market price.
  */
-router.post('/update/price', ensureAuthenticated('manager'), (req: express.Request, res: express.Response) => {
+router.put('/price', ensureAuthenticated('manager'), (req: express.Request, res: express.Response) => {
```
Comment (Aleman778) [inline simulator/src/api/marketapi.ts]: You do not need express.Request/ express.Response type due to type inference.

Comment (Blinningjr) [reply simulator/src/api/marketapi.ts]: It still works, it is only forms in html that can't do put calls

**Pull request merged**